script.js

```diff
@@ -17,3 +17,40 @@ function toggleMenu() {
 
 menu.addEventListener('click', toggleMenu);
 hamburger.addEventListener('click', toggleMenu);
+
+const openModalButtons = document.querySelectorAll('[data-modal-target]');
+const closeModalButtons = document.querySelectorAll('[data-close-button]');
+const overlay = document.getElementById('overlay');
+
+openModalButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const modal = document.querySelector(button.dataset.modalTarget);
+    openModal(modal);
+  });
+});
+
+overlay.addEventListener('click', () => {
+  const modals = document.querySelectorAll('.modal.active');
+  modals.forEach((modal) => {
+    closeModal(modal);
+  });
+});
+
+closeModalButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const modal = button.closest('.modal');
+    closeModal(modal);
+  });
+});
+
+function openModal(modal) {
+  if (modal == null) return;
+  modal.classList.add('active');
+  overlay.classList.add('active');
+}
+
+function closeModal(modal) {
+  if (modal == null) return;
+  modal.classList.remove('active');
+  overlay.classList.remove('active');
+}
```

Comment:
The modal opens and close functionalities are working fine but you are missing out on key requirements for this project.

- [ ] It would be nice if you create an array of objects holding data for each project per object.
- [ ] Kindly clear all previous HTML static data you had, then fetch the data for each project from the array on page load in order to have dynamic data fetched for each project.

**Hint:**
```mdx
const projects = [{
    name: "Title of Project 1",
    description: "Description of project 1",
    image: "link_to_image.jpg",
    technologies: ["HTML", "JavaScript", "Ruby"],
    link: "link_to_live_version.com",
}]
```
- The above is just an example of what the array will be like. You can place the data that each project has.

Reply:
Done###